inference-engine/src/transformations/src/transformations/depth_to_space_fusion.cpp

```diff
@@ -0,0 +1,164 @@
+        // disable the transformation for all plugins, except CPU
+        if (!transformation_callback(std::make_shared<ngraph::opset3::DepthToSpace>())) {
+            return false;
+        }
+
+        auto reshape_after = std::dynamic_pointer_cast<ngraph::opset3::Reshape>(m.get_match_root());
+        if (!reshape_after) {
+            return false;
+        }
+
+        auto permute = std::dynamic_pointer_cast<ngraph::opset3::Transpose>(reshape_after->input_value(0).get_node_shared_ptr());
+        if (!permute) {
+            return false;
+        }
+
+        auto reshape_before = std::dynamic_pointer_cast<ngraph::opset3::Reshape>(permute->input_value(0).get_node_shared_ptr());
+        if (!reshape_before) {
+            return false;
+        }
+
+        auto p_shape_input = reshape_before->get_input_partial_shape(0);
+        auto p_shape_reshape_before = reshape_before->get_output_partial_shape(0);
+        auto p_shape_permute = permute->get_output_partial_shape(0);
+        auto p_shape_reshape_after = reshape_after->get_output_partial_shape(0);
+
+        if (p_shape_input.is_dynamic() || p_shape_reshape_before.is_dynamic() ||
+                p_shape_permute.is_dynamic() || p_shape_reshape_after.is_dynamic()) {
+            return false;
+        }
+
+        auto shape_input = p_shape_input.get_shape();
+        auto shape_reshape_before = p_shape_reshape_before.get_shape();
+        auto shape_permute = p_shape_permute.get_shape();
+        auto shape_reshape_after = p_shape_reshape_after.get_shape();
+
+        if (shape_input.size() < 3) {
+            return false;
+        }
+
+        size_t expected_shape_size = (shape_input.size() - 2) * 2 + 2;
+        if (shape_input.size() != shape_reshape_after.size() || shape_reshape_before.size() != expected_shape_size ||
+            shape_permute.size() != expected_shape_size) {
+            return false;
+        }
+
+        ngraph::AxisVector permutation;
+        if (auto input_const = as_type_ptr<op::Constant>(permute->input_value(1).get_node_shared_ptr())) {
+            permutation = input_const->get_axis_vector_val();
+        } else {
+            return false;
+        }
+
+        ngraph::opset3::DepthToSpace::DepthToSpaceMode mode;
+        size_t block_size;
+        if (check_depth_first(shape_input, shape_reshape_before, permutation, shape_reshape_after, block_size)) {
+            mode = ngraph::opset3::DepthToSpace::DepthToSpaceMode::DEPTH_FIRST;
+        } else if (check_block_first(shape_input, shape_reshape_before, permutation, shape_reshape_after, block_size)) {
+            mode = ngraph::opset3::DepthToSpace::DepthToSpaceMode::BLOCKS_FIRST;
+        } else {
+            return false;
+        }
+
+        auto depth_to_space =
+                std::make_shared<ngraph::opset3::DepthToSpace>(reshape_before->input_value(0).get_node_shared_ptr(), mode, block_size);
+        depth_to_space->set_friendly_name(reshape_before->get_friendly_name());
```

Comment:
I think reshape_after is the correct because it's the last operation in the pattern.
```suggestion
        depth_to_space->set_friendly_name(reshape_after->get_friendly_name());
```